2-Authorization-I/1-call-graph/SPA/src/app/profile/profile.component.ts

```diff
@@ -9,41 +13,67 @@ import { Profile } from '../profile';
   styleUrls: ['./profile.component.css'],
 })
 export class ProfileComponent implements OnInit {
+  profile!: Profile;
   displayedColumns: string[] = ['claim', 'value'];
   dataSource: any = [];
 
   constructor(
     private graphService: GraphService,
-  ) { }
+    private authService: MsalService
+  ) {}
 
   ngOnInit() {
+    const providerOptions: ProviderOptions = {
+      account: this.authService.instance.getActiveAccount()!,
+      scopes: protectedResources.graphMe.scopes,
+      interactionType: InteractionType.Redirect,
+      endpoint: protectedResources.graphMe.endpoint,
+    };
+
+    this.getProfile(providerOptions);
+  }
+
+  getProfile(providerOptions: ProviderOptions) {
     this.graphService
-      .getData(protectedResources.graphMe.endpoint)
+      .getGraphClient(providerOptions)
+      .api('/me')
+      .responseType(ResponseType.RAW)
+      .get()
+      .then((response: any) => {
+        if (response.status === 200) return response.json();
+        if (response.status === 401) {
+          if (response.headers.get('www-authenticate')) {
+            this.graphService.handleClaimsChallenge(response, providerOptions);
+          }
+        }
+      })
       .then((profileResponse: Profile) => {
         this.dataSource = [
           {
             id: 1,
             claim: 'Name',
-            value: profileResponse ? profileResponse.givenName : null,
+            value: profileResponse ? profileResponse['givenName'] : null,
```

Comment:
try using the Profile type to access properties of this response (dot notation), instead of using strings e.g. profileResponse['givenName']

Reply:
Also, let's just display whatever the object has, instead of selecting certain properties out of it